compiler/optimizer/OSRDefAnalysis.cpp

```diff
@@ -1507,6 +1509,7 @@ int32_t TR_OSRLiveRangeAnalysis::fullAnalysis(bool includeParms, bool containsPe
             && (comp()->isOSRTransitionTarget(TR::preExecutionOSR) || comp()->requiresAnalysisOSRPoint(node)))
             {
             TR_ByteCodeInfo bcInfo = node->getByteCodeInfo();
+            TR_ASSERT_FATAL(bcInfo.getCallerIndex() == callerIndex, "Found node with invalid caller index");
             TR_OSRPoint *osrPoint = comp()->getMethodSymbol()->findOSRPoint(bcInfo);
             TR_ASSERT(osrPoint != NULL, "Cannot find a pre OSR point for node %p", node);
```

Comment:
This should be an ASSERT_FATAL as well no? to continue with an osrPoint will just result in a segv….